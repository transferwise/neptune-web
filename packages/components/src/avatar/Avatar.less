@import (reference) '@transferwise/neptune-css/src/variables/neptune-tokens.less';

@avatar-diameter-lg: 72px;
@avatar-diameter-md: 48px;
@avatar-diameter-sm: 24px;

.tw-avatar {
  position: relative;
  border-radius: 50%;
  user-select: none;
  box-sizing: border-box;
  cursor: default;

  & &__content {
    display: flex;
    justify-content: center;
    align-items: center;
    border-radius: 50%;
    overflow: hidden;
    background-color: var(--color-background-neutral);
    height: 100%;
    width: 100%;
    max-height: 100%;
    max-width: 100%;
  }

  &--outlined {
    border: 1px solid var(--color-control-accent);
    cursor: pointer;

    &:hover {
      border-color: var(--color-control-accent-hover);
    }
  }

  &--dark &__content {
    background: @color-base-navy-mid;
    color: @color-text-control;
  }

  &--thumbnail {
    & * {
      width: 100%;
      height: 100%;
      object-fit: cover;
    }
  }

  &--icon {
    & * {
      max-height: 100%;
      max-width: 100%;
    }
  }

  &.tw-avatar--initials,
  &.tw-avatar--icon {
    color: @color-text-primary;
  }

  &.tw-avatar--initials,
  &.tw-avatar--emoji {
    // There is a problem how emojis & text vertically aligned between Windows and Mac. Setting font family to Arial seems to resolve it.
    font-family: Arial, sans-serif;
  }

  &--sm {
    .avatar-dimension(@avatar-diameter-sm, 1px);
  }

  &--md {
    .avatar-dimension(@avatar-diameter-md, 2px);
  }

  &--lg {
    .avatar-dimension(@avatar-diameter-lg, 3px);
  }

  img {
    width: 100%;
    height: 100%;
  }
}

.avatar-dimension(@avatar-size, @mask-size) {
  min-width: @avatar-size;
  width: @avatar-size;
  height: @avatar-size;

  &.tw-avatar--emoji,
  &.tw-avatar--icon {
    font-size: @avatar-size * 0.5;
  }

  &.tw-avatar--initials {
    font-size: @avatar-size * 0.35;
    font-weight: bold;
  }

  &.tw-avatar--outlined {
    padding: @mask-size;

    &.tw-avatar--thumbnail::after {
      top: @mask-size;
      left: @mask-size;
      right: @mask-size;
      bottom: @mask-size;
    }
  }

  &.tw-avatar--thumbnail::after {
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    border: 1px solid rgba(0, 0, 0, 0.08);
    content: '';
    border-radius: 50%;
  }

  &.tw-avatar--dark.tw-avatar--thumbnail::after {
    border-color: rgba(255, 255, 255, 0.08);
  }
}
