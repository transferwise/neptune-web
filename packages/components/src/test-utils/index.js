import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Provider } from '..';
import { DEFAULT_LOCALE } from '../common/locale';
import en from '../../i18n/en.json';

/**
 * Custom `render` function which wraps passed elements in Provider component
 * Source: https://testing-library.com/docs/example-react-intl/#creating-a-custom-render-function
 */
function customRender(ui, { locale = DEFAULT_LOCALE, messages = en, ...renderOptions } = {}) {
  // eslint-disable-next-line react/prop-types
  function Wrapper({ children }) {
    return <Provider i18n={{ locale, messages }}>{children}</Provider>;
  }
  return render(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react';
export { userEvent };
export { customRender as render };
