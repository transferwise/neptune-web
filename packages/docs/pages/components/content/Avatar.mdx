import { LiveEditorBlock, GeneratePropsTable } from '../../../utils';
import { Avatar } from '@transferwise/components';
import { Profile } from '@transferwise/icons';
import code from '../../../liveEditorCode/avatar.code';

<LiveEditorBlock code={code} scope={{ Avatar }} />
<GeneratePropsTable componentName="Avatar" />

<table className="table table-condensed docs-table m-t-5">
  <thead>
    <tr>
      <th>Avatar type</th>
      <th>Avatar size</th>
      <th>Default</th>
      <th>Outline</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Avatar.Type.THUMBNAIL</td>
      <td>Avatar.Type.LARGE</td>
      <td>
        <Avatar type={Avatar.Type.THUMBNAIL} size={Avatar.Size.LARGE}>
          <img
            src="https://transferwise.com/public-resources/assets/flags/square/gbp.svg"
            alt="UK Flag"
          />
        </Avatar>
      </td>
      <td>
        <Avatar type={Avatar.Type.THUMBNAIL} size={Avatar.Size.LARGE} outlined>
          <img
            src="https://transferwise.com/public-resources/assets/flags/square/gbp.svg"
            alt="UK Flag"
          />
        </Avatar>
      </td>
    </tr>
    <tr>
      <td>Avatar.Type.THUMBNAIL</td>
      <td>Avatar.Type.MEDIUM</td>
      <td>
        <Avatar type={Avatar.Type.THUMBNAIL} size={Avatar.Size.MEDIUM}>
          <img
            src="https://transferwise.com/public-resources/assets/flags/square/gbp.svg"
            alt="UK Flag"
          />
        </Avatar>
      </td>
      <td>
        <Avatar type={Avatar.Type.THUMBNAIL} size={Avatar.Size.MEDIUM} outlined>
          <img
            src="https://transferwise.com/public-resources/assets/flags/square/gbp.svg"
            alt="UK Flag"
          />
        </Avatar>
      </td>
    </tr>
    <tr>
      <td>Avatar.Type.THUMBNAIL</td>
      <td>Avatar.Type.SMALL</td>
      <td>
        <Avatar type={Avatar.Type.THUMBNAIL} size={Avatar.Size.SMALL}>
          <img
            src="https://transferwise.com/public-resources/assets/flags/square/gbp.svg"
            alt="UK Flag"
          />
        </Avatar>
      </td>
      <td>
        <Avatar type={Avatar.Type.THUMBNAIL} size={Avatar.Size.SMALL} outlined>
          <img
            src="https://transferwise.com/public-resources/assets/flags/square/gbp.svg"
            alt="UK Flag"
          />
        </Avatar>
      </td>
    </tr>
    <tr>
      <td>Avatar.Type.EMOJI</td>
      <td>Avatar.Type.LARGE</td>
      <td>
        <Avatar size={Avatar.Size.LARGE} type={Avatar.Type.EMOJI} backgroundColor="#D6F5E3">
          <img
            style={{ width: 44, height: 44 }}
            src="https://transferwise.com/public-resources/assets/balances/savings-emoji/google/ðŸŒ±.png"
            alt=""
          />
        </Avatar>
      </td>
      <td>
        <Avatar
          size={Avatar.Size.LARGE}
          type={Avatar.Type.EMOJI}
          backgroundColor="#D6F5E3"
          outlined
        >
          <img
            style={{ width: 44, height: 44 }}
            src="https://transferwise.com/public-resources/assets/balances/savings-emoji/google/ðŸŒ±.png"
            alt=""
          />
        </Avatar>
      </td>
    </tr>
    <tr>
      <td>Avatar.Type.EMOJI</td>
      <td>Avatar.Type.MEDIUM</td>
      <td>
        <Avatar size={Avatar.Size.MEDIUM} type={Avatar.Type.EMOJI} backgroundColor="#D6F5E3">
          <img
            style={{ width: 30, height: 30 }}
            src="https://transferwise.com/public-resources/assets/balances/savings-emoji/google/ðŸŒ±.png"
            alt=""
          />
        </Avatar>
      </td>
      <td>
        <Avatar
          size={Avatar.Size.MEDIUM}
          type={Avatar.Type.EMOJI}
          backgroundColor="#D6F5E3"
          outlined
        >
          <img
            style={{ width: 30, height: 30 }}
            src="https://transferwise.com/public-resources/assets/balances/savings-emoji/google/ðŸŒ±.png"
            alt=""
          />
        </Avatar>
      </td>
    </tr>
    <tr>
      <td>Avatar.Type.EMOJI</td>
      <td>Avatar.Type.SMALL</td>
      <td>
        <Avatar size={Avatar.Size.SMALL} type={Avatar.Type.EMOJI} backgroundColor="#D6F5E3">
          <img
            style={{ width: 30, height: 30 }}
            src="https://transferwise.com/public-resources/assets/balances/savings-emoji/google/ðŸŒ±.png"
            alt=""
          />
        </Avatar>
      </td>
      <td>
        <Avatar
          size={Avatar.Size.SMALL}
          type={Avatar.Type.EMOJI}
          backgroundColor="#D6F5E3"
          outlined
        >
          <img
            style={{ width: 16, height: 16 }}
            src="https://transferwise.com/public-resources/assets/balances/savings-emoji/google/ðŸŒ±.png"
            alt=""
          />
        </Avatar>
      </td>
    </tr>
    <tr>
      <td>Avatar.Type.INITIALS</td>
      <td>Avatar.Type.LARGE</td>
      <td>
        <Avatar type={Avatar.Type.INITIALS} size={Avatar.Size.LARGE}>
          TW
        </Avatar>
      </td>
      <td>
        <Avatar type={Avatar.Type.INITIALS} size={Avatar.Size.LARGE} outlined>
          TW
        </Avatar>
      </td>
    </tr>
    <tr>
      <td>Avatar.Type.INITIALS</td>
      <td>Avatar.Type.MEDIUM</td>
      <td>
        <Avatar type={Avatar.Type.INITIALS} size={Avatar.Size.MEDIUM}>
          TW
        </Avatar>
      </td>
      <td>
        <Avatar type={Avatar.Type.INITIALS} size={Avatar.Size.MEDIUM} outlined>
          TW
        </Avatar>
      </td>
    </tr>
    <tr>
      <td>Avatar.Type.INITIALS</td>
      <td>Avatar.Type.SMALL</td>
      <td>
        <Avatar type={Avatar.Type.INITIALS} size={Avatar.Size.SMALL}>
          TW
        </Avatar>
      </td>
      <td>
        <Avatar type={Avatar.Type.INITIALS} size={Avatar.Size.SMALL} outlined>
          TW
        </Avatar>
      </td>
    </tr>
    <tr>
      <td>Avatar.Type.INITIALS</td>
      <td>Avatar.Type.MEDIUM</td>
      <td>
        <Avatar theme={Avatar.Theme.DARK} type={Avatar.Type.INITIALS} size={Avatar.Size.MEDIUM}>
          TW
        </Avatar>
      </td>
      <td>
        <Avatar theme={Avatar.Theme.DARK} type={Avatar.Type.INITIALS} size={Avatar.Size.MEDIUM} outlined>
          <Profile />
        </Avatar>
      </td>
    </tr>
    <tr>
      <td>Avatar.Type.ICON</td>
      <td>Avatar.Type.LARGE</td>
      <td>
        <Avatar type={Avatar.Type.ICON} size={Avatar.Size.LARGE}>
          <svg width="25" height="25" viewBox="0 0 25 25" fill="currentColor">
            <path d="M22.003 9.408l-10-7.405-10 7.405 1.195 1.595 8.805-6.52 8.805 6.52 1.195-1.595z"></path>
            <path d="M13 10v10h4v-7h2v7h2v2H3v-2h2v-7h2v7h4V10h2z"></path>
          </svg>
        </Avatar>
      </td>
      <td>
        <Avatar type={Avatar.Type.ICON} size={Avatar.Size.LARGE} outlined>
          <svg width="25" height="25" viewBox="0 0 25 25" fill="currentColor">
            <path d="M22.003 9.408l-10-7.405-10 7.405 1.195 1.595 8.805-6.52 8.805 6.52 1.195-1.595z"></path>
            <path d="M13 10v10h4v-7h2v7h2v2H3v-2h2v-7h2v7h4V10h2z"></path>
          </svg>
        </Avatar>
      </td>
    </tr>
    <tr>
      <td>Avatar.Type.ICON</td>
      <td>Avatar.Type.MEDIUM</td>
      <td>
        <Avatar type={Avatar.Type.ICON} size={Avatar.Size.MEDIUM}>
          <svg width="25" height="25" viewBox="0 0 25 25" fill="currentColor">
            <path d="M22.003 9.408l-10-7.405-10 7.405 1.195 1.595 8.805-6.52 8.805 6.52 1.195-1.595z"></path>
            <path d="M13 10v10h4v-7h2v7h2v2H3v-2h2v-7h2v7h4V10h2z"></path>
          </svg>
        </Avatar>
      </td>
      <td>
        <Avatar type={Avatar.Type.ICON} size={Avatar.Size.MEDIUM} outlined>
          <svg width="25" height="25" viewBox="0 0 25 25" fill="currentColor">
            <path d="M22.003 9.408l-10-7.405-10 7.405 1.195 1.595 8.805-6.52 8.805 6.52 1.195-1.595z"></path>
            <path d="M13 10v10h4v-7h2v7h2v2H3v-2h2v-7h2v7h4V10h2z"></path>
          </svg>
        </Avatar>
      </td>
    </tr>
    <tr>
      <td>Avatar.Type.ICON</td>
      <td>Avatar.Type.SMALL</td>
      <td>
        <Avatar type={Avatar.Type.ICON} size={Avatar.Size.SMALL}>
          <svg width="25" height="25" viewBox="0 0 25 25" fill="currentColor">
            <path d="M22.003 9.408l-10-7.405-10 7.405 1.195 1.595 8.805-6.52 8.805 6.52 1.195-1.595z"></path>
            <path d="M13 10v10h4v-7h2v7h2v2H3v-2h2v-7h2v7h4V10h2z"></path>
          </svg>
        </Avatar>
      </td>
      <td>
        <Avatar type={Avatar.Type.ICON} size={Avatar.Size.SMALL} outlined>
          <svg width="25" height="25" viewBox="0 0 25 25" fill="currentColor">
            <path d="M22.003 9.408l-10-7.405-10 7.405 1.195 1.595 8.805-6.52 8.805 6.52 1.195-1.595z"></path>
            <path d="M13 10v10h4v-7h2v7h2v2H3v-2h2v-7h2v7h4V10h2z"></path>
          </svg>
        </Avatar>
      </td>
    </tr>
  </tbody>
</table>

export const meta = {
  name: 'Avatar',
};
