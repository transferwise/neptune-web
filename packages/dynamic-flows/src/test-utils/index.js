import React from 'react';
import { mount } from 'enzyme';
import { Provider } from '@transferwise/components';
import { act } from 'react-dom/test-utils';

const getMockFetchPromise = (status, jsonFn, delay, signal) => {
  const response = {
    status,
    json: jsonFn,
  };
  return new Promise((resolve) => {
    let aborted = false;
    if (signal) {
      signal.addEventListener('abort', () => {
        aborted = true;
      });
    }
    setTimeout(() => {
      return !aborted && resolve(response);
    }, delay);
  });
};

const wait = (t) => {
  return act(() => {
    return new Promise((resolve) => setTimeout(resolve, t));
  });
};

function mountWithProviders(component, locale = 'en-GB', messages = {}) {
  return mount(<Provider i18n={{ locale, messages }}>{component}</Provider>);
}

export { mountWithProviders as mount, getMockFetchPromise, wait };
