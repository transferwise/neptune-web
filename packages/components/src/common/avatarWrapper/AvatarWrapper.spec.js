import React from 'react';
import { render } from '../../test-utils';

import Avatar from '.';

describe('FlowNavigationAvatar', () => {
  it('renders as BUSINESS profile type', async () => {
    const { container } = render(<Avatar url="https://wise.com" profileType="BUSINESS" />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders as BUSINESS profile type with an icon', async () => {
    const { container } = render(<Avatar url="" profileType="BUSINESS" />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders as PERSONAL profile type', async () => {
    const { container } = render(<Avatar url="https://wise.com" profileType="PERSONAL" />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders as PERSONAL profile type with an icon', async () => {
    const { container } = render(<Avatar profileType="PERSONAL" />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
