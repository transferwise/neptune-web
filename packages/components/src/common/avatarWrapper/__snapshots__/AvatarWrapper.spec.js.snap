// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlowNavigationAvatar renders as BUSINESS profile type 1`] = `
<div
  class="tw-avatar tw-avatar--md tw-avatar--thumbnail tw-avatar--light"
>
  <div
    class="tw-avatar__content"
  >
    <img
      alt="avatar"
      src="https://wise.com"
    />
  </div>
</div>
`;

exports[`FlowNavigationAvatar renders as BUSINESS profile type with an icon 1`] = `
<div
  class="tw-avatar tw-avatar--md tw-avatar--icon tw-avatar--light"
>
  <div
    class="tw-avatar__content"
  >
    <span
      aria-hidden="true"
      class="tw-icon tw-icon-briefcase "
      role="presentation"
    >
      <svg
        fill="currentColor"
        focusable="false"
        height="16"
        width="16"
      >
        <path
          d="M7.5.2A2.8 2.8 0 004.7 3h1.6a1.2 1.2 0 011.2-1.2h1A1.2 1.2 0 019.7 3h1.6A2.8 2.8 0 008.5.2h-1z"
        />
        <path
          clip-rule="evenodd"
          d="M15.8 4.2H.2v11.6h15.6V4.2zm-14 10v-3.525c.044.048.089.095.134.14.953.954 1.942.985 2.066.985h2.5v.8h3v-.8c.844 0 1.69.02 2.534 0a1.803 1.803 0 00.208-.02c.124-.018.289-.051.48-.115a3.447 3.447 0 001.344-.85 4.22 4.22 0 00.134-.14V14.2H1.8zm7.7-4h2.484a1.165 1.165 0 00.231-.053c.178-.059.442-.185.72-.463.493-.493 1.126-1.567 1.245-3.884H1.82c.12 2.317.752 3.39 1.246 3.884.462.463.899.512.948.516H6.5v-.8h3v.8z"
          fill-rule="evenodd"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`FlowNavigationAvatar renders as PERSONAL profile type 1`] = `
<div
  class="tw-avatar tw-avatar--md tw-avatar--thumbnail tw-avatar--light"
>
  <div
    class="tw-avatar__content"
  >
    <img
      alt="avatar"
      src="https://wise.com"
    />
  </div>
</div>
`;

exports[`FlowNavigationAvatar renders as PERSONAL profile type with an icon 1`] = `
<div
  class="tw-avatar tw-avatar--md tw-avatar--icon tw-avatar--light"
>
  <div
    class="tw-avatar__content"
  >
    <span
      aria-hidden="true"
      class="tw-icon tw-icon-profile "
      role="presentation"
    >
      <svg
        fill="currentColor"
        focusable="false"
        height="16"
        width="16"
      >
        <path
          clip-rule="evenodd"
          d="M8 0C5.847 0 4.102 1.723 4.102 3.848S5.847 7.696 8 7.696c2.152 0 3.897-1.723 3.897-3.848S10.152 0 8 0zM5.743 3.848c0-1.23 1.01-2.228 2.257-2.228 1.246 0 2.256.998 2.256 2.228 0 1.23-1.01 2.228-2.256 2.228-1.246 0-2.257-.997-2.257-2.228z"
          fill-rule="evenodd"
        />
        <path
          d="M8 9.127c2.487 0 4.604.858 5.967 1.595 1.391.752 2.033 2.22 2.033 3.644V16h-1.641v-1.634c0-.981-.435-1.82-1.18-2.223-1.225-.662-3.065-1.396-5.179-1.396-2.114 0-3.954.734-5.179 1.396-.745.402-1.18 1.242-1.18 2.223V16H0v-1.634c0-1.424.642-2.892 2.033-3.644C3.396 9.985 5.513 9.127 8 9.127z"
        />
      </svg>
    </span>
  </div>
</div>
`;
